### PR TITLE
[Azure Pipeline] switch to Hosted Ubuntu 1604

### PR DESCRIPTION
**Pull request by guolinke**
refer to https://docs.microsoft.com/zh-cn/azure/devops/pipelines/agents/hosted?view=vsts&tabs=yaml#hosted-linux-preview-pool-deprecation

> We're deprecating the Hosted Linux Preview pool. We recommend that you begin moving your pipelines to the Hosted Ubuntu 1604 pool now. We plan to remove the Hosted Linux Preview from Azure Pipelines on December 1, 2018.

### CODE REVIEW

File: .vsts-ci.yml [hunk 1]
```diff
@@ -30,11 +31,12 @@ phases:
...
         PYTHON_VERSION: 3.6
   steps:
-  - task: CondaEnvironment@0
+  - task: CondaEnvironment@1
     inputs:
+      createCustomEnvironment: true
       updateConda: true
       environmentName: $(CONDA_ENV)
-      packageSpecs: 'python=$(PYTHON_VERSION)'
+      packageSpecs: 'python=$(PYTHON_VERSION) numpy nose scipy scikit-learn pandas matplotlib python-graphviz pytest'
```
[the remark this answers is not in the document]
Comment (guolinke) [reply .vsts-ci.yml]: yeah, I forget there is a `if-else` in azure pipeline.

**Comment by StrikerRUS** [comment]
That's for sure! I think hardcoding paths is not good too. Let me fork this branch and try one thing.

**Comment by guolinke** [comment]
@StrikerRUS I think it is fixed now.

**Comment by StrikerRUS** [comment]
@guolinke Sorry, I was from my phone and didn't notice green CI checks.

**Comment by StrikerRUS** [comment]
@guolinke 
From that link you provided in the first post I learned how to switch Xcode version:
```
/bin/bash -c "sudo xcode-select -s /Applications/Xcode_8.3.3.app/Contents/Developer"
```
https://docs.microsoft.com/zh-cn/azure/devops/pipelines/agents/hosted?view=vsts&tabs=yaml#xcode

Versions are listed here: https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/macos/macos-Readme.md#xcode

Maybe this will be useful in the future.

**Comment by guolinke** [comment]
@StrikerRUS awesome! Then, we can try the minimal xcode version for LightGBM.

**Pull request merged**